File: kora-app-annotation-processor/src/test/java/io/koraframework/kora/app/annotation/processor/app/AppWithFactories6.java
package io.koraframework.kora.app.annotation.processor.app;

import io.koraframework.common.KoraApp;
import io.koraframework.common.annotation.Root;

@KoraApp
public interface AppWithFactories6 {

    default <T> GenericClass<T> factory1(GenericClass<T> genericClass) {
        return new GenericClass<>();
    }

    @Root
    default Class2 class2(GenericClass<Class1> class1) {
        return new Class2();
    }


    final class GenericClass<T> {}

    class Class1 {}

    class Class2 {}
}

File: kora-app-symbol-processor/src/test/kotlin/io/koraframework/kora/app/ksp/app/AppWithFactories6.kt
package io.koraframework.kora.app.ksp.app

import io.koraframework.common.KoraApp
import io.koraframework.common.annotation.Root

@KoraApp
interface AppWithFactories6 {
    fun <T> factory1(genericClass: GenericClass<T>): GenericClass<T> {
        return GenericClass()
    }

    @Root
    fun class2(class1: GenericClass<Class1>): Class2 {
        return Class2()
    }

    class GenericClass<T>
    class Class1
    class Class2
}
